### PR TITLE
Fix for missed user.buyerUID

### PR DESCRIPTION
**Pull request by VeronikaSolovei9**

### CODE REVIEW

File: exchange/utils_test.go [hunk 1]
```diff
@@ -3004,6 +3005,13 @@ func TestApplyFPD(t *testing.T) {
...
 			expectedRequest: openrtb2.BidRequest{Site: &openrtb2.Site{ID: "SiteId"}, App: &openrtb2.App{ID: "AppId"}},
 		},
+		{
+			description:     "req.User is defined; bidderFPD.User defined; req.User has BuyerUID. Expect to see user.BuyerUID in result request",
+			inputFpd:        fpdBidderNotNilFPD,
+			bidderName:      bidderNotNilFPD,
+			inputRequest:    openrtb2.BidRequest{User: &openrtb2.User{ID: "UserIdIn", BuyerUID: "12345"}},
+			expectedRequest: openrtb2.BidRequest{User: &openrtb2.User{ID: "UserId", BuyerUID: "12345"}, Site: &openrtb2.Site{ID: "SiteId"}, App: &openrtb2.App{ID: "AppId"}},
+		},
 	}
```
Comment (AlexBVolcy) [inline exchange/utils_test.go:3015]: Could you add another test case for when req.User is defined and bidderFPD.user is defined, but BuyerUID length is 0?

**Review by AlexBVolcy — COMMENTED** [review]
Looks good, close to approving, just think one more test case could be nice.

File: exchange/utils_test.go (hunk 1 above)
Comment (VeronikaSolovei9) [reply exchange/utils_test.go:3010]: Added!

File: exchange/utils_test.go [hunk 2]
```diff
@@ -599,6 +599,7 @@ func TestCleanOpenRTBRequestsWithFPD(t *testing.T) {
 				assert.Equal(t, fpd[bidderName].Site.Name, bidderRequest.BidRequest.Site.Name, "Incorrect FPD site name")
 				assert.Equal(t, fpd[bidderName].App.Name, bidderRequest.BidRequest.App.Name, "Incorrect FPD app name")
 				assert.Equal(t, fpd[bidderName].User.Keywords, bidderRequest.BidRequest.User.Keywords, "Incorrect FPD user keywords")
+				assert.Equal(t, fpd[bidderName].User.BuyerUID, bidderRequest.BidRequest.User.BuyerUID, "Incorrect FPD user buyerUID")
```
[the remark this answers is not in the document]
Comment (VeronikaSolovei9) [reply exchange/utils_test.go]: Agree, modified

File: exchange/utils_test.go [hunk 3]
```diff
@@ -3004,6 +3005,13 @@ func TestApplyFPD(t *testing.T) {
 			inputRequest:    openrtb2.BidRequest{Site: &openrtb2.Site{ID: "SiteId"}, App: &openrtb2.App{ID: "TestAppId"}},
 			expectedRequest: openrtb2.BidRequest{Site: &openrtb2.Site{ID: "SiteId"}, App: &openrtb2.App{ID: "AppId"}},
 		},
+		{
```
[the remark this answers is not in the document]
Comment (VeronikaSolovei9) [reply exchange/utils_test.go:2992]: Added

**Comment by snigel-nelson** [comment]
Hi,
It seems the problem is solved and the solution is merged and available in the [Release v0.214.0](https://github.com/prebid/prebid-server/releases/tag/v0.214.0).
Regarding this issue: https://github.com/prebid/prebid-server/issues/2238 I'll update my PBS to v0.214.0 and see if it works. If everything is OK I can close it :)
Cheers

**Pull request merged**